controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesReaderGroup.java

```diff
@@ -156,7 +156,11 @@ private String getMetadataTableName(String id) {
 
     @Override
     public CompletableFuture<Void> delete() {
-        return getId().thenCompose(id -> Futures.toVoid(storeHelper.deleteTable(getMetadataTableName(id), false)));
+        return getId().thenCompose(id -> storeHelper.deleteTable(getMetadataTableName(id), false)
+        .thenCompose(v -> {
+            this.idRef = new AtomicReference<>(null);
```

Comment:
actually refresh the idRef in the "refresh" method of this class. otherwise this delete could have happened on a different controller while this controller actually continues to use a cached object which a stale idref.

Reply:
done